indexer/osm_editor.cpp

```diff
@@ -678,21 +689,31 @@ NewFeatureCategories Editor::GetNewFeatureCategories() const
   return res;
 }
 
-namespace
-{
-FeatureID GenerateNewFeatureId(FeatureID const & oldFeatureId)
+FeatureID Editor::GenerateNewFeatureId(MwmSet::MwmId const & id)
 {
-  // TODO(AlexZ): Stable & unique features ID generation.
-  // TODO(vng): Looks like new feature indexes should uninterruptedly continue after existing indexes in mwm file.
-  static uint32_t newIndex = 0x0effffff;
-  return FeatureID(oldFeatureId.m_mwmId, newIndex++);
+  // TODO(vng): Double-check if new feature indexes should uninterruptedly continue after existing indexes in mwm file.
+  uint32_t featureIndex = kStartIndexForCreatedFeatures;
+  auto const found = m_features.find(id);
+  if (found != m_features.end())
+  {
+    // Scan all already created features and choose next available ID.
+    for (auto const & feature : found->second)
```

Comment:
Didn’t get the question. What does ‘sorted by id’ mean?
New features will have some ids, like 0xffff0001, 0xffff0002 etc.

What do you have in mind?